skimage/draw/_random_shapes.py

```diff
@@ -402,22 +404,23 @@ def random_shapes(image_shape,
     num_shapes = random.randint(min_shapes, max_shapes + 1)
     colors = _generate_random_colors(num_shapes, num_channels,
                                      intensity_range, random)
+    shape = (min_size, max_size)
     for shape_idx in range(num_shapes):
         if user_shape is None:
             shape_generator = random.choice(SHAPE_CHOICES)
         else:
             shape_generator = SHAPE_GENERATORS[user_shape]
-        shape = (min_size, max_size)
         for _ in range(num_trials):
             # Pick start coordinates.
-            column = random.randint(image_shape[1])
-            row = random.randint(image_shape[0])
+            column = random.randint(max(1, image_shape[1] - min_size))
+            row = random.randint(max(1, image_shape[0] - min_size))
```

Comment:
Why do you start at 1 and not 0?

Reply:
when only the `low` parameter is provided to [`randint`](https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.random.randint.html) it acts as `high` :upside_down_face:

Reply:
Bleugh. I misread. LOL My preference would be to add a 0 as `low` anyway to make things explicit and easier to parse. But I am open to being ignored on this particular point. 😂